client/my-sites/stats/stats-email-detail/index.jsx

```diff
@@ -140,6 +148,23 @@ class StatsEmailDetail extends Component {
 
 	componentDidMount() {
 		window.scrollTo( 0, 0 );
+
+		const { context } = this.props;
+		recordCurrentScreen( 'postDetails', {
+			queryParams: context.query,
+			period: null,
+		} );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { context } = this.props;
+
+		if ( ! isEqual( prevProps.context, this.props.context ) ) {
+			recordCurrentScreen( 'postDetails', {
```

Comment:
Does it deserve an `emailDetails` object key?
```suggestion
			recordCurrentScreen( 'emailDetails', {
```